ai/extract_features.py

```
@@ -146,18 +147,14 @@
-    )
+    output_path = os.path.join(OUTPUT_DIR, f"{category}_features_tfidf_{n_components}.csv")
     result.to_csv(output_path, index=False)
     print(f"Saved features to {output_path}")
 
     return result
 
 
-def _extracted_from_extract_features_73(
-    df, vectorizer, category, n_components, random_state
-):
+def _extracted_from_extract_features_73(df, vectorizer, category, n_components, random_state):
```

Pylint (3.3.7)

[refactor] Too many local variables (22/15)

(R0914)